### PR TITLE
Add explicit LICENSE files (and Cargo.toml `license`)

### PR DESCRIPTION
**Pull request by tuommaki**

### CODE REVIEW

File: LICENSE-APACHE
Comment (tuommaki) [inline LICENSE-APACHE:1]: This is sourced from GitHub license template.

File: LICENSE-APACHE [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2023 Gevulot Oy
```
Comment (tuommaki) [inline LICENSE-APACHE:189]: :raised_hands:

File: LICENSE-MIT
Comment (tuommaki) [inline LICENSE-MIT:1]: Same as above. Sourced from GitHub license template.

File: LICENSE-MIT [hunk 2]
```diff
@@ -0,0 +1,22 @@
+MIT License
+
+Copyright (c) 2023 Gevulot Oy
```
Comment (tuommaki) [inline LICENSE-MIT:3]: :rocket:

File: README.md
Comment (tuommaki) [inline README.md:1]: Let's reference our own files, shall we? :grin:

File: prover/Cargo.toml [hunk 3]
```diff
@@ -2,6 +2,7 @@
 name = "prover"
 version = "0.3.0"
 edition = "2021"
+license = "MIT or Apache-2.0"
```
Comment (tuommaki) [inline prover/Cargo.toml:5]: This helps with the tooling.

**Review by teempai — APPROVED** [review]
Looks good

**Pull request merged**